explorer/client/src/components/view-more/ViewMore.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { useState, useCallback } from 'react';
+
+import styles from './ViewMore.module.css';
+
+type Props = {
+    children: JSX.Element[];
+    limitTo?: number;
+    label?: string;
+};
+
+const DEFAULT_MIN_NUMBER_OF_ITEMS_TO_SHOW = 3;
+
+function ViewMore({ children, limitTo, label }: Props) {
+    const [viewMore, setVeiwMore] = useState(false);
+    const numberOfListItemsToShow =
+        limitTo || DEFAULT_MIN_NUMBER_OF_ITEMS_TO_SHOW;
+    const viewAll = useCallback(() => {
+        setVeiwMore(!viewMore);
```

Comment:
Can be ` setVeiwMore((prev) => !prev);`, and then you can avoid the hook dependency.